docs/src/snap/howto/backup-restore.md

```diff
@@ -0,0 +1,304 @@
+# Backup and restore
+
+[Velero][] is a popular open source backup solution for Kubernetes.
+Its core implementation is a controller running in the cluster that oversees
+the backup and restore operations. The administrator is given a CLI tool to
+schedule operations and/or perform on-demand backup and restores. This CLI
+tool creates Kubernetes resources that the in-cluster Velero controller acts
+upon. During installation the controller needs to 
+be [configured with a repository (called a 'provider')][providers], where
+the backup files are stored.
+
+This document describes how to setup Velero with the MinIO provider acting
+as an S3 compatible object store.
+
+
+## What you will need
+
+- A running Canonical Kubernetes with DNS enabled
+- MinIO (install described below)
+- Velero (install described below))
+- An example workload
+
+### Enabling required components
+
+DNS is needed for this setup:
+
+```bash
+sudo k8s enable dns
+```
+
+### Install MinIO
+
+[MinIO][] provides an S3 compatible interface over storage provisioned by
+Kubernetes. For the purposes of this guide, the `local storage` component
+is used to satisfy the persistent volume claims:
+
+```bash
+sudo k8s enable local-storage
+```
+
+Helm is used to setup [MinIO][MinIO Charts] under the `velero` namespace:
+
+```bash
+sudo k8s helm repo add minio https://charts.min.io
+sudo k8s kubectl create namespace velero
+
+# TOY MODE
+sudo k8s helm install -n velero \
+--set resources.requests.memory=512Mi \
+--set replicas=1 \
+--set persistence.enabled=false \
+--set mode=standalone \
+--set buckets[0].name=velero,buckets[0].policy=none,buckets[0].purge=false \
+--generate-name minio/minio
+
+# PRODUCTION MODE
+sudo k8s helm install -n velero \
+--set buckets[0].name=velero,buckets[0].policy=none,buckets[0].purge=false \
+--generate-name minio/minio
+```
+
+### Create a demo workload
+
+The workload we will demonstrate the backup with is an NGINX deployment and a
+corresponding service under the `workloads` namespace. Create this setup with:
+
+```bash
+sudo k8s kubectl create namespace workloads
+sudo k8s kubectl create deployment nginx -n workloads --image nginx
+sudo k8s kubectl expose deployment nginx -n workloads --port 80
+```
+
+
+## Install Velero
+
+Download the Velero binary from the 
+[releases page on github][releases] and place it in our `PATH`. In this case we
+install the v1.14.1 Linux binary for AMD64 under `/usr/local/bin`:
+
+```bash
+wget https://github.com/vmware-tanzu/velero/releases/download/v1.14.0/velero-v1.14.0-linux-amd64.tar.gz 
+tar -xzf velero-v1.14.0-linux-amd64.tar.gz
+chmod +x velero-v1.14.0-linux-amd64/velero
+sudo chown root:root velero-v1.14.0-linux-amd64/velero
+sudo mv velero-v1.14.0-linux-amd64/velero /usr/local/bin/velero
+```
+
+Before installing Velero, we export the kubeconfig file using the `config`
```

Comment:
Maybe this step should go first then?

Reply:
I think i needed to reword this a bit.  The steps were right. 

there's "install a tool" and "install into the cluster"